### PR TITLE
Refactor CMakeLists.txt

### PR DESCRIPTION
**Pull request by anutosh491**
Removed redundancy and adds more consistency especially for the wasm build

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -98,31 +100,6 @@ if (NOT TARGET xeus AND NOT TARGET xeus-static)
...
-    if(NOT EMSCRIPTEN)
-        add_compile_options(-Wunused-parameter -Wextra -Wreorder)
-    endif()
-endif ()
-
-if(EMSCRIPTEN)
-    # ENV (https://github.com/emscripten-core/emscripten/commit/6d9681ad04f60b41ef6345ab06c29bbc9eeb84e0)
-    set(EMSCRIPTEN_FEATURES "${EMSCRIPTEN_FEATURES} -s \"EXTRA_EXPORTED_RUNTIME_METHODS=[ENV']\"")
-endif()
-
```
Comment (anutosh491) [inline CMakeLists.txt:125]: These are redundant as are being addressed through `macro(xeus_cpp_set_common_options target_name)` so can be totally removed from here.

**Comment by anutosh491** [comment]
@vgvassilev this should be ready (P.S i made the exact same change on xeus-r few days ago and I realized they would be applicable on xeus-cpp too so pretty confident :) )

File: CMakeLists.txt (hunk 1 above)
Comment (JohanMabille) [reply CMakeLists.txt:125]: I think some flags are still missing in the macro (-fexceptions and /EHsc, I may have missed others).

Comment (anutosh491) [reply CMakeLists.txt:125]: Hmm, having a look !

Comment (anutosh491) [reply CMakeLists.txt:125]: Yupp those were the only ones. Should be ready now !

File: CMakeLists.txt [hunk 2]
```diff
@@ -347,7 +332,6 @@ macro(xeus_cpp_create_target target_name linkage output_name)
         # Curl initialised specifically for xassist
         target_link_libraries(${target_name} PUBLIC ${XEUS_CPP_XEUS_TARGET} clangCppInterOp pugixml argparse::argparse curl)
     else ()
-        # TODO : Add curl support for emscripten
```
Comment (anutosh491) [inline CMakeLists.txt:350]: Removed TODO which can't be resolved as mentioned here (https://github.com/emscripten-forge/recipes/pull/1332#issuecomment-2371509339)

**Review by JohanMabille — APPROVED** [review]

**Pull request merged**